pkg/build/registry/buildconfig/strategy.go

```diff
@@ -21,10 +21,14 @@ import (
 type strategy struct {
 	runtime.ObjectTyper
 	names.NameGenerator
+	legacy bool
 }
 
 // Strategy is the default logic that applies when creating and updating BuildConfig objects.
-var Strategy = strategy{kapi.Scheme, names.SimpleNameGenerator}
+var Strategy = strategy{kapi.Scheme, names.SimpleNameGenerator, false}
+
+// LegacyStrategy is the default logic that applies when creating and updating BuildConfig objects.
```

Comment:
Should this comment and the one for Strategy be more explanatory? Like, why would one be applied over the other?  As they stand it sounds like they are both applied by default?

Also, s/applies/is applied/ ?